File: backend/models/awards.py
"""
Awards Models
End-of-year award categories and winners.
"""

from dataclasses import dataclass
from typing import List, Dict, Any, Optional
from enum import Enum


class AwardCategory(Enum):
    """Award categories for end-of-year ceremony"""
    # Wrestler Awards
    WRESTLER_OF_THE_YEAR = "wrestler_of_the_year"
    BREAKOUT_STAR = "breakout_star"
    MOST_IMPROVED = "most_improved"
    COMEBACK_OF_THE_YEAR = "comeback_of_the_year"
    
    # Match Awards
    MATCH_OF_THE_YEAR = "match_of_the_year"
    FEUD_OF_THE_YEAR = "feud_of_the_year"
    MOMENT_OF_THE_YEAR = "moment_of_the_year"
    
    # Championship Awards
    CHAMPION_OF_THE_YEAR = "champion_of_the_year"
    TITLE_REIGN_OF_THE_YEAR = "title_reign_of_the_year"
    
    # Tag Team Awards
    TAG_TEAM_OF_THE_YEAR = "tag_team_of_the_year"
    TAG_MATCH_OF_THE_YEAR = "tag_match_of_the_year"
    
    # Performance Awards
    BEST_TECHNICAL_WRESTLER = "best_technical_wrestler"
    BEST_HIGH_FLYER = "best_high_flyer"
    BEST_BRAWLER = "best_brawler"
    
    # Microphone Awards
    PROMO_OF_THE_YEAR = "promo_of_the_year"
    BEST_ON_THE_MIC = "best_on_the_mic"
    
    # Negative Awards
    BIGGEST_DISAPPOINTMENT = "biggest_disappointment"
    WORST_FEUD = "worst_feud"
    
    # Special Awards
    BEST_BRAND = "best_brand"
    SHOW_OF_THE_YEAR = "show_of_the_year"


@dataclass
class AwardNominee:
    """A nominee for an award"""
    nominee_id: str
    nominee_name: str
    nominee_type: str  # 'wrestler', 'match', 'show', 'feud', 'tag_team'
    stats: Dict[str, Any]
    reason: str
    score: float
    
    def to_dict(self) -> Dict[str, Any]:
        return {
            'nominee_id': self.nominee_id,
            'nominee_name': self.nominee_name,
            'nominee_type': self.nominee_type,
            'stats': self.stats,
            'reason': self.reason,
            'score': round(self.score, 2)
        }


@dataclass
class Award:
    """An award with nominees and winner"""
    category: AwardCategory
    year: int
    nominees: List[AwardNominee]
    winner_id: str
    winner_name: str
    
    def to_dict(self) -> Dict[str, Any]:
        return {
            'category': self.category.value,
            'category_display': self._get_display_name(),
            'year': self.year,
            'nominees': [n.to_dict() for n in self.nominees],
            'winner_id': self.winner_id,
            'winner_name': self.winner_name
        }
    
    def _get_display_name(self) -> str:
        """Convert enum to display name"""
        name_map = {
            AwardCategory.WRESTLER_OF_THE_YEAR: "Wrestler of the Year",
            AwardCategory.BREAKOUT_STAR: "Breakout Star of the Year",
            AwardCategory.MOST_IMPROVED: "Most Improved Wrestler",
            AwardCategory.COMEBACK_OF_THE_YEAR: "Comeback of the Year",
            AwardCategory.MATCH_OF_THE_YEAR: "Match of the Year",
            AwardCategory.FEUD_OF_THE_YEAR: "Feud of the Year",
            AwardCategory.MOMENT_OF_THE_YEAR: "Moment of the Year",
            AwardCategory.CHAMPION_OF_THE_YEAR: "Champion of the Year",
            AwardCategory.TITLE_REIGN_OF_THE_YEAR: "Title Reign of the Year",
            AwardCategory.TAG_TEAM_OF_THE_YEAR: "Tag Team of the Year",
            AwardCategory.TAG_MATCH_OF_THE_YEAR: "Tag Team Match of the Year",
            AwardCategory.BEST_TECHNICAL_WRESTLER: "Best Technical Wrestler",
            AwardCategory.BEST_HIGH_FLYER: "Best High-Flyer",
            AwardCategory.BEST_BRAWLER: "Best Brawler",
            AwardCategory.PROMO_OF_THE_YEAR: "Promo of the Year",
            AwardCategory.BEST_ON_THE_MIC: "Best on the Mic",
            AwardCategory.BIGGEST_DISAPPOINTMENT: "Biggest Disappointment",
            AwardCategory.WORST_FEUD: "Worst Feud of the Year",
            AwardCategory.BEST_BRAND: "Brand of the Year",
            AwardCategory.SHOW_OF_THE_YEAR: "Show of the Year"
        }
        return name_map.get(self.category, self.category.value.replace('_', ' ').title())


@dataclass
class AwardsCeremony:
    """Complete awards ceremony for a year"""
    year: int
    awards: List[Award]
    ceremony_date_week: int
    
    def to_dict(self) -> Dict[str, Any]:
        return {
            'year': self.year,
            'ceremony_date_week': self.ceremony_date_week,
            'total_awards': len(self.awards),
            'awards': [a.to_dict() for a in self.awards]
        }

File: frontend/static/js/awards.js
/**
 * Awards Management
 */

let allCeremonies = [];
let currentCeremony = null;

// Load on page load
document.addEventListener('DOMContentLoaded', function() {
    loadAllAwards();
});

// ============================================================================
// LOAD AWARDS
// ============================================================================

async function loadAllAwards() {
    try {
        const response = await getAPI('/api/awards/all');
        
        if (response.success) {
            allCeremonies = response.ceremonies;
            populateYearSelect();
            
            if (allCeremonies.length > 0) {
                // Load most recent ceremony
                loadCeremony(allCeremonies[0]);
            }
        }
        
    } catch (error) {
        console.error('Failed to load awards:', error);
        showEmptyState('No awards ceremonies found');
    }
}

function populateYearSelect() {
    const select = document.getElementById('yearSelect');
    
    if (allCeremonies.length === 0) {
        select.innerHTML = '<option value="">No awards yet - Complete Year 1!</option>';
        return;
    }
    
    let options = '';
    
    allCeremonies.forEach((ceremony, index) => {
        const selected = index === 0 ? 'selected' : '';
        options += `<option value="${ceremony.year}" ${selected}>Year ${ceremony.year} Awards</option>`;
    });
    
    select.innerHTML = options;
    
    // Add event listener
    select.addEventListener('change', function() {
        const year = parseInt(this.value);
        const ceremony = allCeremonies.find(c => c.year === year);
        if (ceremony) {
            loadCeremony(ceremony);
        }
    });
}

function loadCeremony(ceremony) {
    currentCeremony = ceremony;
    
    // Update summary
    document.getElementById('awardsSummary').style.display = 'block';
    document.getElementById('summaryYear').textContent = ceremony.year;
    document.getElementById('summaryTotal').textContent = ceremony.total_awards;
    
    // Render awards
    renderAwards(ceremony.awards);
}

// ============================================================================
// RENDER AWARDS
// ============================================================================

function renderAwards(awards) {
    const container = document.getElementById('awardsContainer');
    
    if (!awards || awards.length === 0) {
        showEmptyState('No awards for this year');
        return;
    }
    
    let html = '';
    
    // Group awards by type
    const wrestlerAwards = awards.filter(a => 
        a.category.includes('wrestler') || 
        a.category.includes('breakout') || 
        a.category.includes('improved') ||
        a.category.includes('comeback')
    );
    
    const matchAwards = awards.filter(a => 
        a.category.includes('match') || 
        a.category.includes('feud')
    );
    
    const championshipAwards = awards.filter(a => 
        a.category.includes('champion') || 
        a.category.includes('reign')
    );
    
    const tagAwards = awards.filter(a => 
        a.category.includes('tag')
    );
    
    const performanceAwards = awards.filter(a => 
        a.category.includes('technical') || 
        a.category.includes('flyer') || 
        a.category.includes('brawler') ||
        a.category.includes('mic')
    );
    
    const showAwards = awards.filter(a => 
        a.category.includes('show') || 
        a.category.includes('brand')
    );
    
    // Render each group
    if (wrestlerAwards.length > 0) {
        html += '<h3 class="mt-4 mb-3"><i class="bi bi-person-badge"></i> Wrestler Awards</h3>';
        wrestlerAwards.forEach(award => {
            html += renderAwardCard(award);
        });
    }
    
    if (championshipAwards.length > 0) {
        html += '<h3 class="mt-4 mb-3"><i class="bi bi-trophy"></i> Championship Awards</h3>';
        championshipAwards.forEach(award => {
            html += renderAwardCard(award);
        });
    }
    
    if (matchAwards.length > 0) {
        html += '<h3 class="mt-4 mb-3"><i class="bi bi-lightning"></i> Match & Feud Awards</h3>';
        matchAwards.forEach(award => {
            html += renderAwardCard(award);
        });
    }
    
    if (tagAwards.length > 0) {
        html += '<h3 class="mt-4 mb-3"><i class="bi bi-people"></i> Tag Team Awards</h3>';
        tagAwards.forEach(award => {
            html += renderAwardCard(award);
        });
    }
    
    if (performanceAwards.length > 0) {
        html += '<h3 class="mt-4 mb-3"><i class="bi bi-star"></i> Performance Awards</h3>';
        performanceAwards.forEach(award => {
            html += renderAwardCard(award);
        });
    }
    
    if (showAwards.length > 0) {
        html += '<h3 class="mt-4 mb-3"><i class="bi bi-tv"></i> Show Awards</h3>';
        showAwards.forEach(award => {
            html += renderAwardCard(award);
        });
    }
    
    container.innerHTML = html;
}

function renderAwardCard(award) {
    const isPrestigious = award.category.includes('wrestler_of_the_year') || 
                          award.category.includes('match_of_the_year');
    
    const headerClass = isPrestigious ? 'award-category-header gold' : 'award-category-header';
    
    let html = `
        <div class="award-category-card">
            <div class="${headerClass}">
                <div class="award-icon">${getAwardIcon(award.category)}</div>
                <div class="award-category-title">${award.category_display}</div>
            </div>
            
            <div class="award-winner-section">
                <div class="winner-badge">
                    <i class="bi bi-trophy-fill"></i>
                    WINNER
                </div>
                <div class="winner-name">${award.winner_name}</div>
                <div class="winner-reason">${award.nominees[0].reason}</div>
            </div>
            
            ${award.nominees.length > 1 ? `
                <div class="nominees-section">
                    <h6 class="text-muted mb-3">Other Nominees</h6>
                    ${award.nominees.slice(1).map((nominee, index) => `
                        <div class="nominee-item">
                            <div class="nominee-rank">${index + 2}</div>
                            <div class="nominee-info">
                                <div class="nominee-name">${nominee.nominee_name}</div>
                                <div class="nominee-reason">${nominee.reason}</div>
                            </div>
                            <div class="nominee-score">${nominee.score.toFixed(1)}</div>
                        </div>
                    `).join('')}
                </div>
            ` : ''}
        </div>
    `;
    
    return html;
}

function getAwardIcon(category) {
    const icons = {
        'wrestler_of_the_year': '👑',
        'breakout_star': '🌟',
        'most_improved': '📈',
        'comeback_of_the_year': '🔄',
        'match_of_the_year': '⚡',
        'feud_of_the_year': '🔥',
        'moment_of_the_year': '💫',
        'champion_of_the_year': '🏆',
        'title_reign_of_the_year': '👑',
        'tag_team_of_the_year': '👥',
        'tag_match_of_the_year': '⚔️',
        'best_technical_wrestler': '🎯',
        'best_high_flyer': '🦅',
        'best_brawler': '👊',
        'promo_of_the_year': '🎤',
        'best_on_the_mic': '🎙️',
        'best_brand': '📺',
        'show_of_the_year': '🎪'
    };
    
    return icons[category] || '🏆';
}

function showEmptyState(message) {
    const container = document.getElementById('awardsContainer');
    container.innerHTML = `
        <div class="empty-awards">
            <i class="bi bi-trophy"></i>
            <h4>${message}</h4>
            <p>Awards ceremonies appear at the end of each year (Week 52)</p>
        </div>
    `;
}